src/components/HoursModal/HoursModal.tsx

```diff
@@ -0,0 +1,283 @@
+import { useEffect, useState } from 'react';
+import { Button, Col, Row } from 'react-bootstrap';
+import Modal from 'react-bootstrap/Modal';
+import Switch from '../../components/Switch/Switch';
+import TimePicker from 'rc-time-picker';
+import { useAppSelector } from '../../redux/store';
+import { saveDateRequest } from '../../api/reservationDeskBackend/calendarApi';
+import isAfter from 'date-fns/isAfter';
+import moment from 'moment';
+import { format } from 'date-fns';
+
+type HoursModalProps = {
+  show: boolean;
+  close: () => void;
+  shownImageMap: string;
+  currDate: string;
+  deskId: string;
+  reservedHours: string;
+  checkIfReservationIsSaved: (isSaved: string) => void;
+};
+
+const HoursModal = ({
+  show,
+  close,
+  shownImageMap,
+  currDate,
+  deskId,
+  reservedHours,
+  checkIfReservationIsSaved,
+}: HoursModalProps) => {
+  const [timeValueFrom, setTimeValueFrom] = useState<Date | number>(0);
+  const [timeValueTo, setTimeValueTo] = useState<Date | number>(0);
+  const [isTimepickerVisible, setIsTimepickerVisible] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+  const [isReservationOptionChecked, setIsReservationOptionChecked] =
+    useState(false);
+
+  const userId = useAppSelector((state) => state.user.userId);
+
+  //enable/disable switches, check if user chose option before save, control visibility of timepickers
+  const handleOnClickSwitch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.target.checked && errorMsg ? setErrorMsg('') : null;
+
+    setIsReservationOptionChecked(event.target.checked);
+
+    event.target.checked
+      ? document
+          .querySelector('.switch-btn input:not(#' + event.target.id + ')')
+          ?.setAttribute('disabled', 'disabled')
+      : document
+          .querySelector('.switch-btn input:not(#' + event.target.id + ')')
+          ?.removeAttribute('disabled');
+
+    setIsTimepickerVisible(
+      event.target.getAttribute('id') == 'timepicker-switch'
+        ? event.target.checked
+        : false
+    );
+  };
+
+  //check if the reservation is not overlapping another one
+  const isValidReservationTime = (startTime: Date, endTime: Date) => {
+    const disabledHoursArr = disabledHours();
+    let startHour = 0;

```

Comment:
Why not assign the value of the startHour variable the same line when you initialize it?